### ISSUE TITLE
Build breaks because of missing declaration

### ISSUE THREAD

**Issue by tuxology** [iovisor/llvm_bpf#1]
Running `make` under the `bld` directory throws the following error:

```
llvm[3]: Compiling BPFCFGFixup.cpp for Debug+Asserts build
<path_to>/llvm_bpf/lib/Target/BPF/BPFCFGFixup.cpp:60:59: 
error: ‘llvm::FunctionPass* llvm::createBPFCFGFixup(llvm::BPFTargetMachine&)’ 
should have been declared inside ‘llvm’
```

I am on Fedora 20 with clang version 3.4. Here is a tiny patch which fixes this:

```
diff --git a/lib/Target/BPF/BPF.h b/lib/Target/BPF/BPF.h
index 53fe49a..3a61077 100644
--- a/lib/Target/BPF/BPF.h
+++ b/lib/Target/BPF/BPF.h
@@ -21,7 +21,7 @@ class BPFTargetMachine;
 /// createBPFISelDag - This pass converts a legalized DAG into a
 /// BPF-specific DAG, ready for instruction scheduling.
 FunctionPass *createBPFISelDag(BPFTargetMachine &TM);
-
+FunctionPass *createBPFCFGFixup(BPFTargetMachine &TM);
 extern Target TheBPFTarget;
 }
```

**Comment by iovisor-obsolete** [comment]
In the last commit I removed all callers of this pass, but forgot to remove the file itself.
Just pushed the fix. Thanks for the report.
btw, BPF backend has been upstreamed. Please try the latest llvm trunk.

**Comment by iovisor-obsolete** [comment]
fixed by commit cd57510157
